plutus-core/plutus-core/src/PlutusCore/TypeCheck.hs

```diff
@@ -25,23 +27,20 @@ import PlutusPrelude
 
 import PlutusCore.Builtin
 import PlutusCore.Core
-import PlutusCore.Error
 import PlutusCore.Name
 import PlutusCore.Normalize
 import PlutusCore.Quote
 import PlutusCore.Rename
 import PlutusCore.TypeCheck.Internal
 
-import Control.Monad.Except
-import Data.Array
 import Universe
 
 type Typecheckable uni fun = (ToKind uni, HasUniApply uni, ToBuiltinMeaning uni fun)
```

Comment:
Hmm. Isn't this paritally redundant with the new constraints that you added? should they be consolidated?

Reply:
There's a small overlap, yes. I think it's only `ToKind uni`.

I did try removing it when I was refactoring the code, but it's used in multiple places downstream and some of them don't have any other type checking related constraints. This is because while we really only need `Typecheckable` for, well, type checking, all the other type checking related constraints talk about the monad to run type checking in, which occasionally gets instantiated, which still leaves `Typecheckable`. For example

```haskell
lift
    :: (Lift.Lift uni a, Throwable uni fun, PLC.Typecheckable uni fun)
    => a -> UPLC.Term UPLC.NamedDeBruijn uni fun ()
```

I could try to remove the `ToKind uni` overlap, but I believe it's totally not worth introducing yet another type-level entity for that.

I'll add a comment.

Reply:
> I'm also gonna try removing `ToKind uni` from `Typecheckable`, maybe I just missed it could be done.

Tried it and it was a mess, so I'm keeping `ToKind uni` duplicated between the two constraints.

Documented the whole thing.